Melt/identify_species.py

```diff
@@ -0,0 +1,137 @@
+        end = first + seconds_per_sample
+        return liklihood, first + 1, end
+    elif count == 2:
+        max_liklihood = max(liklihoods[first:last + 1])
+        min_liklihood = min(liklihoods[first:last + 1])
+        if max_liklihood == liklihoods[first + 1]:
+            start = first + 1
+            end = start + seconds_per_sample
+            return max_liklihood, start, end
+        elif min_liklihood == liklihoods[first]:
+            start = first + 1
+            end = last + seconds_per_sample
+            return max_liklihood, start, end
+        elif min_liklihood == liklihoods[last]:
+            end = first + 1 + seconds_per_sample
+            return max_liklihood, first, end
+        else:
+            end = last + seconds_per_sample
+            return max_liklihood, first, end
+    else:
+        max_liklihood = max(liklihoods[first:last + 1])
+        if max_liklihood > liklihoods[first]:
+            start = first + 1
+            if max_liklihood > liklihoods[last]:
+                end = last - 1 + seconds_per_sample
+                return max_liklihood, start, end
+            else:
+                end = last + seconds_per_sample
+                return max_liklihood, start, end
+        elif max_liklihood > liklihoods[last]:
+            end = last - 1 + seconds_per_sample
+            return max_liklihood, first, end
+        else:
+            start = first + 1
+            end = last - 1 + seconds_per_sample
+            return max_liklihood, start, end
+
+
+def identify_species(recording, metadata, models):
+
+    # get spectrogram to be checked
+    sr, npspec = _load_sample(recording)
+
+    # divide recording into samples of appropriate length
+    samples = []
+    for base in range(0, npspec.shape[1], sample_slide_slices):
+        limit = base + slices_per_sample
+        if limit > npspec.shape[1]:
+            limit = npspec.shape[1]
+        start = limit - slices_per_sample
+        sample = npspec[:, start:limit]
+        sample = librosa.power_to_db(sample, ref=np.max)
+        sample = sample / abs(sample.min()) + 1.0
+        samples.append(sample.reshape(sample.shape + (1,)))
+    samples = np.array(samples)
+
+    # accumulate results from all models
+    activations_sum = np.zeros(len(samples))
+    model_paths = _model_paths(models)
+    for path in model_paths:
+        model = tf.keras.models.load_model(path)
+        activations = model.predict(samples).flatten()
+        activations_sum += activations
+
+    def entry(begin, end, species, activation):
+        entry = {}
+        entry['begin_s'] = begin * sample_slide_seconds
+        entry['end_s'] = (i - 1) * sample_slide_seconds + seconds_per_sample
+        entry['species'] = species
+        entry['liklihood'] = round(activation * 0.01, 2)
+        return entry
+
+    # generate labels from summed activations
+    labels = []
+    liklihoods = [round(v * 33.33333333) for v in activations_sum]
```

Comment:
`v * 100 / 3.0` guessing this is because there is 3 models?

Reply:
yes, made this dependency on the number of models explict